openshift_cli_installer/libs/clusters/ipi_cluster.py

```diff
@@ -89,43 +92,62 @@ def _create_install_config_file(self):
             "pull_secret": self.pull_secret,
         }
 
-        worker_flavor = self.cluster.get("worker-flavor")
-        if worker_flavor:
+        if worker_flavor := self.cluster.get("worker-flavor"):
             terraform_parameters["worker_flavor"] = worker_flavor
 
-        worker_root_disk_size = self.cluster.get("worker-root-disk-size")
-        if worker_root_disk_size:
+        if worker_root_disk_size := self.cluster.get("worker-root-disk-size"):
             terraform_parameters["worker_root_disk_size"] = worker_root_disk_size
 
-        worker_replicas = self.cluster.get("worker-replicas")
-        if worker_replicas:
+        if worker_replicas := self.cluster.get("worker-replicas"):
             terraform_parameters["worker_replicas"] = worker_replicas
 
-        fips = self.cluster.get("fips")
-        if fips:
+        if fips := self.cluster.get("fips"):
             terraform_parameters["fips"] = fips
 
-        cluster_install_config = get_install_config_j2_template(jinja_dict=terraform_parameters)
+        return terraform_parameters
+
+    def _create_install_config_file(self, platform):
+        cluster_install_config = get_install_config_j2_template(jinja_dict=self.terraform_parameters, platform=platform)
 
         with open(os.path.join(self.cluster_info["cluster-dir"], "install-config.yaml"), "w") as fd:
             fd.write(yaml.dump(cluster_install_config))
 
     def _set_install_version_url(self):
         cluster_version = self.cluster["version"]
-        version_url = [url for url, versions in self.aws_base_available_versions.items() if cluster_version in versions]
+        version_url = [url for url, versions in self.ipi_base_available_versions.items() if cluster_version in versions]
         if version_url:
             self.cluster_info["version-url"] = f"{version_url[0]}:{self.cluster_info['version']}"
         else:
             self.logger.error(
                 f"{self.log_prefix}: Cluster version url not found for"
-                f" {cluster_version} in {self.aws_base_available_versions.keys()}",
+                f" {cluster_version} in {self.ipi_base_available_versions.keys()}",
             )
             raise click.Abort()
 
+    def move_gcp_service_account_file(self, file_action):
```

Comment:
```suggestion
    def set_gcp_service_account_file(self, file_action):
```
the function sets up the config

- please add docstring
- split to 2 functions - set and unset; you need to save the state:

set function:
1. if file exists - move to tmp and save provided file in path
2. if file does not exist - save in path
3.  save state

unset function:
if file existed - restore orig file
if not - delete provided file from path